### PR TITLE
Add flag to include config files for diagnostics script

### PR DESCRIPTION
**Pull request by anders-elastisys**
> [!warning]
> **This is a public repository, ensure not to disclose:**
>
> - [x] personal data beyond what is necessary for interacting with this pull request, nor
> - [x] business confidential information, such as customer names.

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [x] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [x] kind/admin-change 
- [ ] kind/dev-change   
- [ ] kind/security     
- [ ] [kind/adr]()      

### Platform Administrator notice

The command usage for `ck8s` has been updated so that all commands are listed in ascending order.

The command `ck8s diagnostics` has a new flag to include config files in the output log file. The command usage has been changed: `ck8s diagnostics <wc|sc> [-n namespace] [--include-config]`

### What does this PR do / why do we need this PR?

This PR adds a new flag for the `ck8s diagnostics` command to include config files for the environment found in `CK8S_CONFIG_PATH`. This PR also adds a flag parser and changes the command usage for specifying a namespace. Also the `ck8s` command usage has been sorted in ascending order as it imo started to feel very cluttered, this way it is a bit easier to follow.

- Part of #2154

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [x] The change is transparent
  - [ ] The change is disruptive
  - [ ] The change requires no migration steps
  - [ ] The change requires migration steps
  - [ ] The change upgrades CRDs
  - [ ] The change updates the config *and* the schema
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [x] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packets in the `NetworkPolicy Dashboard`
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Bug checks:
  - [ ] The bug fix is covered by regression tests

### CODE REVIEW

File: bin/ck8s
Comment (salehsedghpour) [inline bin/ck8s:27]: Do we want to remove such reminders?

File: bin/diagnostics.bash
Comment (salehsedghpour) [inline bin/diagnostics.bash:296]: Does it also work in a multi-wc scenario?

File: bin/ck8s
Comment (anders-elastisys) [reply bin/ck8s:27]: I am unsure, but the comments were 4 year old and personally I don't find them relevant as the `ops` commands are generally super useful as a Platform Administrator, hence I feel its good to keep them listed in the usage. It might however be a good idea to improve the usage, separate and group commands together etc. but that would be a separate issue. I don't have to address the `ck8s` usage here at all, but it might be good to improve it in the future. I would like to hear what others think about changing/updating the `ck8s` usage

File: bin/diagnostics.bash
Comment (anders-elastisys) [reply bin/diagnostics.bash:296]: This would only include config files found in `CK8S_CONFIG_PATH`, so it will only include the `wc` found in the current `CK8S_CONFIG_PATH`. To get config files for another `wc` you would have to change `CK8S_CONFIG_PATH`, atleast with how we do multi-wc atm

File: bin/ck8s
Comment (simonklb) [reply bin/ck8s:27]: A lot has changed in 4 years (I think the comments might actually be even older than that). We are now moving towards making use of existing tooling directly rather than scripting around them. This means that hopefully commands like `ck8s apply` and `ck8s ops helm` will soon just become `helmfile apply` and `helm`.

TL;DR - the comments can be removed. :)

**Review by Elias-elastisys — APPROVED** [review]

File: bin/diagnostics.bash
Comment (salehsedghpour) [reply bin/diagnostics.bash:296]: Great, then I guess we only need to document it in public docs.

Comment (anders-elastisys) [reply bin/diagnostics.bash:296]: Yes, I have a [PR prepared](https://github.com/elastisys/compliantkubernetes/pull/941) here and I will just wait until I merge this one.

**Pull request merged**
